src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -116,24 +124,20 @@ SELECT COUNT(DISTINCT p) FROM StudentParticipation p
 
     /**
      * @param exerciseId id of exercise
-     * @param correctionRound correction round to find completed assessments by
      * @return the number of completed assessments for the specified correction round of an exam exercise
      */
     @Query("""
-            SELECT COUNT(DISTINCT p)
-            FROM StudentParticipation p WHERE p.exercise.id = :exerciseId
-            AND p.testRun = FALSE
-            AND (SELECT COUNT(r)
-                FROM p.results r
-                WHERE r.assessor IS NOT NULL
-                AND r.rated = TRUE
-                AND r.submission = (select max(id) from p.submissions)
-                AND r.submission.submitted = TRUE
+            SELECT COUNT(p.id)
+            FROM StudentParticipation p left join  p.submissions s left join s.results r
```

Comment:
Are you sure you need a left join here?

Reply:
I'm not sure, what difference would it make?

Reply:
Well you seem to obviously want only the participations with submissions and results, so why not filter those without submission and results already in the join condition? This will be faster imo

Reply:
I replaced it, and it still works, so I guess it's a good idea :) faster is always better